### PR TITLE
Add GPT-2 model 

### PR DESCRIPTION
**Pull request by ZiyueXu77**
Fixes # .

### Description

Add GPT-2 model to the NLP example, corrections to the data sequence and label alignment function, update readme 

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [x] Documentation updated.

### CODE REVIEW

File: examples/advanced/sklearn-kmeans/sklearn_kmeans_iris.ipynb [hunk 1]
```diff
@@ -66,12 +66,76 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 6,
    "id": "f4130b15-09e6-456f-a3c7-87c8ee9e07f0",
    "metadata": {
     "tags": []
    },
```
[the remark this answers is not in the document]
Comment (ZiyueXu77) [reply examples/advanced/sklearn-kmeans/sklearn_kmeans_iris.ipynb]: ah thanks! residual from yesterday, I should have used a different folder for that.

**Review by holgerroth — APPROVED** [review]
Looks good.

**Review by holgerroth — CHANGES_REQUESTED** [review]
Change computing of model weights to `model_diff[name] = np.subtract(local_weights[name].cpu().numpy(), global_weights[name], dtype=np.float32)` to ensure the model is runnable on more systems.

**Comment by ZiyueXu77** [comment]
> Change computing of model weights to `model_diff[name] = np.subtract(local_weights[name].cpu().numpy(), global_weights[name], dtype=np.float32)` to ensure the model is runnable on more systems.

was planning to merge this first then update all in a single PR, anyway, updated this one first

**Review by holgerroth — APPROVED** [review]

**Pull request merged**